src/test/java/com/linkedin/kafka/clients/utils/tests/EmbeddedBrokerBuilder.java

```diff
@@ -0,0 +1,234 @@
+
+
+public class EmbeddedBrokerBuilder {
+  private final static AtomicInteger BROKER_ID_COUNTER = new AtomicInteger();
+
+  //mandatory fields
+  private int nodeId = BROKER_ID_COUNTER.incrementAndGet();
+  private String zkConnect;
+  //storage config
+  private File logDirectory;
+  //networking config
+  private int plaintextPort = -1;
+  private int sslPort = -1;
+  private SecurityProtocol interBrokerSecurity;
+  private File trustStore;
+  private long socketTimeout = 1500;
+  //feature control
+  private boolean enableControlledShutdown;
+  private long controlledShutdownRetryBackoff = 100;
+  private boolean enableDeleteTopic;
+  private boolean enableLogCleaner;
+  //resource management
+  private long logCleanerDedupBufferSize = 2097152; //2MB
+  private String rack;
+
+  //builder state
+  boolean plaintextPortSet = false;
+  boolean sslPortSet = false;
+
+  public EmbeddedBrokerBuilder() {
+  }
+
+  public EmbeddedBrokerBuilder nodeId(int nodeId) {
+    this.nodeId = nodeId;
+    return this;
+  }
+
+  public EmbeddedBrokerBuilder zkConnect(String zkConnect) {
+    this.zkConnect = zkConnect;
+    return this;
+  }
+
+  public EmbeddedBrokerBuilder zkConnect(EmbeddedZookeeper zk) {
+    return zkConnect(zk.getConnectionString());
+  }
+
+  public EmbeddedBrokerBuilder logDirectory(File logDirectory) {
+    this.logDirectory = logDirectory;
+    return this;
+  }
+
+  public EmbeddedBrokerBuilder enable(SecurityProtocol protocol) {
+    switch (protocol) {
+      case PLAINTEXT:
+        enablePlaintext();
+        break;
+      case SSL:
+        enableSsl();
+        break;
+      default:
+        throw new IllegalStateException("unhandled: " + protocol);
+    }
+    return this;
+  }
+
+  public EmbeddedBrokerBuilder plaintextPort(int plaintextPort) {
+    this.plaintextPort = plaintextPort;
+    plaintextPortSet = true;
+    return this;
+  }
+
+  public EmbeddedBrokerBuilder enablePlaintext() {
+    return plaintextPort(0);
+  }
+
+  public EmbeddedBrokerBuilder sslPort(int sslPort) {
+    this.sslPort = sslPort;
+    sslPortSet = true;
+    return this;
+  }
+
+  public EmbeddedBrokerBuilder enableSsl() {
+    return sslPort(0);
+  }
+
+  public EmbeddedBrokerBuilder interBrokerSecurity(SecurityProtocol interBrokerSecurity) {
+    this.interBrokerSecurity = interBrokerSecurity;
+    if (interBrokerSecurity == null) {
+      if (!plaintextPortSet) {
+        plaintextPort = -1;
+      }
+      if (!sslPortSet) {
+        sslPort = -1;
+      }
+    } else {
+      switch (interBrokerSecurity) {
+        case PLAINTEXT:
+          if (!plaintextPortSet) {
+            plaintextPort = 0;
+          }
+          break;
+        case SSL:
+          if (!sslPortSet) {
+            sslPort = 0;
+          }
+          break;
+        default:
+          throw new UnsupportedOperationException("unhandled: " + interBrokerSecurity);
+      }
+    }
+    return this;
+  }
+
+  public EmbeddedBrokerBuilder trustStore(File trustStore) {
```

Comment:
Do the trust stores need to be files or can they be resources so that you can package truststores in the source code.

Reply:
the scala harness uses a File as a trust store, and i matched it.
also, this is test code, i see no use case for packaging trust stores into the artifacts ?